frontend/src/pages/auth/login/index.tsx

```diff
@@ -0,0 +1,215 @@
+import React, { useState } from 'react';
+import { useForm } from '@tanstack/react-form';
+import { isEmail, isEmpty } from 'validator';
+import { authenticationService } from '@/services/auth-service';
+import { writeTokens } from '@/utils/local-storage';
+import { flushSync } from 'react-dom';
+import { jwtDecode, JwtPayload } from 'jwt-decode';
+import { GoogleLogin } from '@react-oauth/google';
+import { IUserDetails, ILoginFormData } from 'shared-types';
+import { useNavigate } from '@tanstack/react-router';
+import {
+  TextField,
+  Grid,
+  Divider,
+  Button,
+  Typography,
+  FormControlLabel,
+  Checkbox,
+  CircularProgress,
+  Alert,
+} from '@mui/material';
+import { useAuth } from '@/utils/auth-context';
+
+const validateEmail = (email: string) => !isEmpty(email) && isEmail(email);
+
+const LoginPage: React.FC = () => {
+  const navigate = useNavigate();
+  const auth = useAuth();
+
+  const [errorOccurred, setErrorOccurred] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const loginForm = useForm<ILoginFormData>({
+    defaultValues: {
+      email: '',
+      password: '',
+      remember: true,
+    },
+    onSubmit: async ({ value }) => {
+      try {
+        setErrorOccurred(false);
+        setIsLoading(true);
+        const { email, password, remember } = value;
+        const tokens = await authenticationService.login(email, password);
+
+        writeTokens(tokens, remember);
+
+        flushSync(() => {
+          const payload = jwtDecode<JwtPayload & IUserDetails>(tokens.accessToken, {});
+          auth.setUser(payload);
+        });
+
+        navigate({ to: '/' });
+      } catch {
+        setErrorOccurred(true);
+      } finally {
+        setIsLoading(false);
+      }
+    },
+    validators: {
+      onSubmit({ value }) {
+        const requiredFields = ['email', 'password'] as Array<keyof typeof value>;
+        if (!validateEmail(value.email) || requiredFields.some((field) => isEmpty(value[field].toString())))
+          return 'Missing or invalid values';
+      },
+    },
+  });
+
+  const handleGoogleSuccess = async (credential: string) => {
+    try {
+      setErrorOccurred(false);
+      setIsLoading(true);
+      const tokens = await authenticationService.googleSignIn(credential);
+      writeTokens(tokens, loginForm.getFieldValue('remember'));
```

Comment:
```suggestion
      writeTokens(tokens, true);
```

Always remember when using google sign in since its not intuitive to click the remember me if using google sign in